### PR TITLE
Update ratings as soon as the user rates a skill (Fixed #1378)

### LINKED ISSUE #1378: Update ratings as soon as the user rates a skill
**Actual Behaviour**

Currently, if the user rates a skill, the data shown in the skill rating section doesn't update. It updates only when the user goes back to the chat activity and returns to the skills listing activity again and checks the skill rating after this. This is because a call to the getSkillList API is sent only when the skills listing activity is opened.

**Expected Behaviour**

The rating shown in the skill details section should be updated as soon as the user rates a skill.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by arundhati24**
Fixes #1378 

A demo for can be seen here : [Link](https://drive.google.com/open?id=1UMlIIG5e-COCVYexxnTxGIGk9wH5_iv0)

### CODE REVIEW

**Comment by arundhati24** [comment]
@iamareebjamal Here are the code changes. Please have a look. Thanks :)

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/FiveStarSkillRatingResponse.kt [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+    @SerializedName("session")
+    @Expose
     val session: Session? = null
+
+    @SerializedName("accepted")
+    @Expose
     val accepted: Boolean = false
+
+    @SerializedName("message")
+    @Expose
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/FiveStarSkillRatingResponse.kt]: I don't think you need SerializedName and Expose if property names are name, please confirm

**Comment by iamareebjamal** [comment]
The logic seems too complicated for a simple use case. Why not use coroutines instead of AsyncTask

File: app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/FiveStarSkillRatingResponse.kt (hunk 1 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/rest/responses/susi/FiveStarSkillRatingResponse.kt]: @iamareebjamal That's right. Thanks. Actually these annotations were not there initially in this file. Now I understand why these were not there. I will remove these as all the variable names used here are same as the keys.

**Comment by arundhati24** [comment]
Okay 👍 I will find out more about coroutines and make the necessary changes soon.

**Comment by arundhati24** [comment]
@iamareebjamal I have used coroutines for the first time. I had referred to the following resources to understand Kotlin coroutines : [Kotlin/kotlinx.coroutines](https://github.com/Kotlin/kotlinx.coroutines/blob/master/ui/coroutines-guide-ui.md#the-problem-of-ui-freezes), [Kotlin/kotlin-coroutines](https://github.com/Kotlin/kotlin-coroutines)  
I had updated this PR but I am unable to get the app running as the CI builds are failing. I am unable to understand it. I might be missing on something small. It would be very helpful if you could please have a look at the code changes and suggest the necessary changes.

**Comment by iamareebjamal** [comment]
`More than one file was found with OS independent path 'META-INF/main.kotlin_module'`

File: app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * @author arundhati24
+ */
+interface ISkillDetailsModel {
+
+    interface OnUpdateRatingsFinishedListener {
+        fun onError(throwable: Throwable)
+        fun onSkillDetailsModelSuccess(response: Response<FiveStarSkillRatingResponse>)
+    }
+
+    fun fiveStarRateSkill(model: String, group: String, language: String, skill: String, stars: String,
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/data/contract/ISkillDetailsModel.kt]: Too many params, make object instead

File: app/src/main/java/org/fossasia/susi/ai/data/SkillDetailsModel.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+     *             model       Model of the skill (e.g. general)
+     *             group       Group of skill (e.g. Knowledge)
+     *             language    Language directory in which the skill resides (e.g. en)
+     *             skill       Skill Tag of object in which the skill data resides
+     *             rating      User rating to be sent to the server
+     *             accessToken Access token of logged in user
+     *             listener
+     *
+     */
+    override fun fiveStarRateSkill(map: Map<String, String>, listener: ISkillDetailsModel.OnUpdateRatingsFinishedListener) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/data/SkillDetailsModel.kt]: Use object, maps aren't typesafe

**Comment by arundhati24** [comment]
Closing this, as this has already been merged along with other PR. Will replace query map with object in a new PR.

**Pull request closed without merging**